imapclient/dovecot_test.go

```diff
@@ -17,13 +17,23 @@ func newDovecotClientServerPair(t *testing.T) (net.Conn, io.Closer) {
 ssl           = no
 mail_home     = "` + tempDir + `/%u"
 mail_location = maildir:~/Mail
+mail_plugins = acl
+protocol imap {
+  mail_plugins = $mail_plugins imap_acl
+}
+plugin {
+  acl = vfile
+}
 `
+
 	if err := os.WriteFile(cfgFilename, []byte(cfg), 0666); err != nil {
 		t.Fatalf("failed to write Dovecot config: %v", err)
 	}
 
 	clientConn, serverConn := net.Pipe()
 
+	// this works for me, while doveadm for some reason don't
+	// cmd := exec.Command("/usr/libexec/dovecot/imap", "-c", cfgFilename)
```

Comment:
I know that it shouldn't be committed, I wanted to ask maybe you could help? Is it working in your environment?(

Reply:
Yeah, it's working fine for me. The CI is using Alpine Linux, and my local setup is Arch Linux.

What is the error you're running into? What is your local OS?

Reply:
I'm using CentOS 7. The problem is that doveadm for some reason does not show the ACL extension, when direct run of imap binary with the same config works. 

```
doveadm -D -c test.conf exec imap
Debug: Loading modules from directory: /usr/lib64/dovecot
Debug: Module loaded: /usr/lib64/dovecot/lib01_acl_plugin.so
Debug: Loading modules from directory: /usr/lib64/dovecot/doveadm
Debug: Module loaded: /usr/lib64/dovecot/doveadm/lib10_doveadm_acl_plugin.so
Debug: Skipping module doveadm_expire_plugin, because dlopen() failed: /usr/lib64/dovecot/doveadm/lib10_doveadm_expire_plugin.so: undefined symbol: expire_set_deinit (this is usually intentional, so just ignore this message)
Debug: Skipping module doveadm_quota_plugin, because dlopen() failed: /usr/lib64/dovecot/doveadm/lib10_doveadm_quota_plugin.so: undefined symbol: quota_user_module (this is usually intentional, so just ignore this message)
Debug: Skipping module doveadm_fts_lucene_plugin, because dlopen() failed: /usr/lib64/dovecot/doveadm/lib20_doveadm_fts_lucene_plugin.so: undefined symbol: lucene_index_iter_deinit (this is usually intentional, so just ignore this message)
Debug: Skipping module doveadm_fts_plugin, because dlopen() failed: /usr/lib64/dovecot/doveadm/lib20_doveadm_fts_plugin.so: undefined symbol: fts_user_get_language_list (this is usually intentional, so just ignore this message)
Debug: Skipping module doveadm_mail_crypt_plugin, because dlopen() failed: /usr/lib64/dovecot/doveadm/libdoveadm_mail_crypt_plugin.so: undefined symbol: mail_crypt_box_get_pvt_digests (this is usually intentional, so just ignore this message)
* PREAUTH [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR LOGIN-REFERRALS ID ENABLE IDLE SORT SORT=DISPLAY THREAD=REFERENCES THREAD=REFS THREAD=ORDEREDSUBJECT MULTIAPPEND URL-PARTIAL CATENATE UNSELECT CHILDREN NAMESPACE UIDPLUS LIST-EXTENDED I18NLEVEL=1 CONDSTORE QRESYNC ESEARCH ESORT SEARCHRES WITHIN CONTEXT=SEARCH LIST-STATUS BINARY MOVE SNIPPET=FUZZY] Logged in as k.roshchina
```

```
/usr/libexec/dovecot/imap -c test.conf
* PREAUTH [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR LOGIN-REFERRALS ID ENABLE IDLE SORT SORT=DISPLAY THREAD=REFERENCES THREAD=REFS THREAD=ORDEREDSUBJECT MULTIAPPEND URL-PARTIAL CATENATE UNSELECT CHILDREN NAMESPACE UIDPLUS LIST-EXTENDED I18NLEVEL=1 CONDSTORE QRESYNC ESEARCH ESORT SEARCHRES WITHIN CONTEXT=SEARCH LIST-STATUS BINARY MOVE SNIPPET=FUZZY ACL RIGHTS=texk] Logged in as k.roshchina
```

Reply:
dovecot version is 2.2.36 (1f10bfa63)

Reply:
So... I've removed dovecot that I installed with yum and builded it from source https://github.com/dovecot/core. And it works now somehow. And also it has more caps by default then dovecot that I installed previously. Tests also succeed now in my environment.

```
doveadm -c test.conf exec imap
Error: conn unix:/usr/local/var/run/dovecot/stats-writer: net_connect_unix(/usr/local/var/run/dovecot/stats-writer) failed: Permission denied
imap(k.roshchina): Error: conn unix:/usr/local/var/run/dovecot/stats-writer: net_connect_unix(/usr/local/var/run/dovecot/stats-writer) failed: Permission denied
* PREAUTH [CAPABILITY IMAP4rev1 SASL-IR LOGIN-REFERRALS ID ENABLE IDLE SORT SORT=DISPLAY THREAD=REFERENCES THREAD=REFS THREAD=ORDEREDSUBJECT MULTIAPPEND URL-PARTIAL CATENATE UNSELECT CHILDREN NAMESPACE UIDPLUS LIST-EXTENDED I18NLEVEL=1 CONDSTORE QRESYNC ESEARCH ESORT SEARCHRES WITHIN CONTEXT=SEARCH LIST-STATUS BINARY MOVE SNIPPET=FUZZY PREVIEW=FUZZY PREVIEW STATUS=SIZE SAVEDATE COMPRESS=DEFLATE INPROGRESS LITERAL+ NOTIFY ACL RIGHTS=texk] Logged in as k.roshchina
```

Reply:
Maybe it's possible to update CI environment as such, so the tests won't fail?